src/bin/pg_autoctl/pgsql.c

```diff
@@ -1253,21 +1253,31 @@ parsePgsrSyncStateAndWAL(void *ctx, PGresult *result)
 
 		case 1:
 		{
-			char *strWalLag = PQgetvalue(result, 0, 1);
+			strlcpy(context->latestLSN,
+					PQgetvalue(result, 0, 1),
+					NAMEDATALEN);
 
 			/* we trust our length and PostgreSQL results */
 			strlcpy(context->syncState,
 					PQgetvalue(result, 0, 0),
 					PGSR_SYNC_STATE_MAXLENGTH);
 
-			context->walLag = strtoull(strWalLag, NULL, 10);
 
-			if (context->walLag == 0 && errno != 0)
+#if 0
```

Comment:
This looks like an attempt to run PostgreSQL backend code in the client application...

Reply:
yes indeed, a failed attempt.